internal/resources/grafana/resource_report_test.go

```diff
@@ -114,6 +116,7 @@ func TestAccResourceReport_basic(t *testing.T) {
 					resource.TestCheckResourceAttr("grafana_report.test", "schedule.0.workdays_only", "true"),
 					resource.TestCheckResourceAttr("grafana_report.test", "schedule.0.start_time", "2020-01-01T07:00:00Z"), // Date transformed to UTC
 					resource.TestCheckResourceAttr("grafana_report.test", "schedule.0.end_time", "2020-01-15T08:30:00Z"),   // Date transformed to UTC
```

Comment:
Do the given times require timezone information? If that's the case, it's not possible to fully use the timezone, right?

Reply:
I'm seeing that the frontend is sending startDate and endDate with the timezone selected.
In the backend, it uses the timezone in the scheduler [here](https://github.com/grafana/grafana-enterprise/blob/main/src/pkg/extensions/report/scheduler/utils.go#L81) copying the location information into the start date, but it isn't applied to the end date. Its working because the dates have the timezone applied in the create/update report calls.

When we retrieve the dashboards we applied the timezone in both values, but internally isn't in both... 

I'll talk with sharing team tomorrow about that

Reply:
I re-checked it again and I'm seeing that currently we need to set the start/end dates with the timezone, but we are forcing `GMT` timezone. The result is that in the backend is using GMT to send the report ignoring the timezone set in the dates (except for the end time like I said before). 

You can see the test code [here](https://github.com/grafana/terraform-provider-grafana/blob/master/examples/resources/grafana_report/all-options.tf#L19-L20). It sets the timezones but in the test result, these values aren't appear because report's backend is applying the timezone for the response. So this test is fine.

I think that because this inconsistency in the end date, some reports aren't finishing in the correct date and it was wrong all the time 😅.

We can do 2 things:
* Add the timezone field like I did and apply it in the star/end dates ignoring the timezone from these fields.
*  Don't add the timezone and extract the timezone from the dates.

WDYT that it's the best way to do it?

Reply:
> Don't add the timezone and extract the timezone from the dates.

I think this one doesn't work because timezone indicators don't include daylight savings, it's just UTC or `[-+]\d`

> Add the timezone field like I did and apply it in the star/end dates ignoring the timezone from these fields.

What about, we combine both of those options:

- If the user has timezone indicators in their dates AND the `timezone` field set, throw an error because they shouldn't be used together
- If the `timezone` is set but the dates don't have timezones to them, use the `timezone` field
- If the `timezone` field isn't set, but the dates have timezones, use those

What do you think?

Reply:
We have two things:
* Timezone is mandatory to send into the reports and we can't leave it empty. (Now its hardcoded to GMT)
* We need to change the validation function from start/end time since we are validating against [RFC3339](https://golang.cafe/blog/golang-time-format-example.html). `DateTime` could be close to what we need.

Reply:
Yeah, we shouldn't require RFC3339 in the dates themselves if we're setting the timezone field. Perhaps a RFC3339 but without the TZ info (just `2006-01-02T15:04:05`)

Reply:
I just updated everything.

I needed to create a custom time format and I set it in the description of the field. If not isn't going to be intuitive enough.